plugins/modules/aci_interface_policy_eigrp.py

```diff
@@ -0,0 +1,370 @@
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),  # Not required for querying all objects
+        eigrp=dict(type="str", aliases=["eigrp_interface", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        bw=dict(type="int"),
+        delay=dict(type="int"),
+        controls=dict(type="list", elements="str", choices=["split-horizon", "bfd", "nh-self", "passive"]),
+        hold_interval=dict(type="int", default="15"),
+        hello_interval=dict(type="int", default="5"),
+        delayUnit=dict(type="str", default="tens-of-micro"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["eigrp", "tenant"]],
+            ["state", "present", ["eigrp", "tenant"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    eigrp = module.params.get("eigrp")
+    description = module.params.get("description")
+    name_alias = module.params.get("name_alias")
+
+    if module.params.get("controls") is None:
+        controls = None
+    else:
+        controls = ",".join(module.params.get("controls"))
+
+    bw = module.params.get("bw")
+    if bw is not None and bw not in range(0, 2560000000):
+        module.fail_json(msg="Parameter 'bw' is only valid in range between 0 and 2560000000.")
+
+    hold_interval = module.params.get("hold_interval")
+    if hold_interval is not None and hold_interval not in range(1, 65536):
+        module.fail_json(msg="Parameter 'dead_interval' is only valid in range between 1 and 65536.")
+
+    hello_interval = module.params.get("hello_interval")
+    if hello_interval is not None and hello_interval not in range(1, 65536):
+        module.fail_json(msg="Parameter 'hello_interval' is only valid in range between 1 and 65536.")
+
+    delay = module.params.get("delay")
+    if delay is not None and delay < 0:
+        module.fail_json(msg="Parameter 'delay' is only valid if 'delay' > 0.")
+
+    delayUnit = module.params.get("delayUnit")
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="eigrpIfPol",
+            aci_rn="tn-{0}/eigrpIfPol-{1}".format(tenant, eigrp),
+            module_object=eigrp,
+            target_filter={"name": eigrp},
+        ),
+    )
```

Comment:
Please follow this sample module to build the url: https://github.com/CiscoDevNet/ansible-aci/blob/86243d26613b15ba0b326f8ac3f0a82408c8d46f/plugins/modules/aci_tenant_span_src_group.py#L255-L269